File: JavaScript/Common_Algorithms/QuickSort/quickSort.js
function quicksort(array) {
  if (array.length == 0) return [];
  let left = [];
  let right = [];
  let pivot = array[0];

  for (var i = 1; i < array.length; i++) {
    if (array[i] < pivot) {
      left.push(array[i]);
    }
    else {
      right.push(array[i])
    }
  }
    return quicksort(left).concat(pivot, quicksort(right));
}

quicksort([5,2,7,9,4])

File: Ruby/Common_Sorting_Algorithms/quick_sort/quickSort.rb
def quicksort(array)
  return [] if array.length == 0
  right = []
  left = []
  pivot = array[0]
  i = 1
  
  while i < array.length
    if array[i] < pivot
      left.push(array[i])
    else
      right.push(array[i])
    end
    i += 1
  end
  sorted_left = quicksort(left)
  sorted_right = quicksort(right)

  return sorted_left + [pivot] + sorted_right

end

p quicksort([5,2,7,9,4])
